### PR TITLE
Add Proper Docstring, Remove Dev settings, and use QgsNetworkManager 

### PR DESCRIPTION
**Pull request by zamuzakki**
This is draft PR for https://github.com/ConservationInternational/cplus-plugin/issues/451

- [x] To download file, we can use https://docs.qgis.org/3.34/en/docs/user_manual/processing_algs/qgis/filetools.html
- [x] For uploading file, currently, we're still using requests.
- [x] There is also still slow performance when updating CplusApiPooling to use QgsNetworkAccessManager

### CODE REVIEW

**Review by danangmassandy — APPROVED** [review]

**Review by Samweli — COMMENTED** [review]
Thanks @zamuzakki and @danangmassandy for the work! Here are the few things I have noted.

1. Usage of non blocking calls from QgsNetworkAccessManager inside QgsTasks. We should use blocking calls eg https://qgis.org/pyqgis/3.38/core/QgsNetworkAccessManager.html#qgis.core.QgsNetworkAccessManager.blockingGet
2. Inside of using the QgsFileDownloader we could leverage and use the `qgis:filedownloader` algorithm from QGIS processing and remove the need of event loop usage.
3. We could have only one API based clase and move the BaseApiClient into the CplusApiRequest.
4. Lets update all the docstrings to have space between each variable description.

File: src/cplus_plugin/utils.py [hunk 1]
```diff
@@ -557,9 +563,14 @@ def default(self, obj):
         return json.JSONEncoder.default(self, obj)
 
 
-def todict(obj, classkey=None):
+def todict(obj: object, classkey: typing.Union[typing.Type, None] = None) -> dict:
     """
     Convert any object to dictionary
+
+    :param obj: Object to be converted
```
Comment (Samweli) [inline src/cplus_plugin/utils.py:570]:
```suggestion
    :param obj: Object to be converted
    :type obj: object
    
    :param classkey: Class to be used key
    :type classkey: typing.Union[typing.Type, None]
```

**Comment by zamuzakki** [comment]
@Samweli I'll update the PR. Out of curiosity, why do we avoid using event loop?

**Comment by zamuzakki** [comment]
Closing this PR by splitting into 3 PRs:
https://github.com/ConservationInternational/cplus-plugin/pull/515
https://github.com/ConservationInternational/cplus-plugin/pull/516
https://github.com/ConservationInternational/cplus-plugin/pull/517

**Pull request closed without merging**